ros2topic/ros2topic/verb/echo.py

```diff
@@ -82,6 +85,61 @@ def add_arguments(self, parser, cli_name):
         parser.add_argument(
             '--raw', action='store_true', help='Echo the raw binary representation')
 
+    def choose_qos(self, node, args):
+
+        if (args.qos_profile or args.qos_reliability or args.qos_durability or
+            args.qos_depth or args.qos_history):
+                return qos_profile_from_short_keys(args.qos_profile,
+                                               reliability=args.qos_reliability,
+                                               durability=args.qos_durability,
+                                               depth=args.qos_depth,
+                                               history=args.qos_history)
+
+        qos_profile = QoSPresetProfiles.get_from_short_key('sensor_data')
+        reliability_reliable_endpoints_count = 0
+        durability_transient_local_endpoints_count = 0
+        publishers_count = 0
+
+        for info in node.get_publishers_info_by_topic(args.topic_name):
+            publishers_count += 1
+            if (info.qos_profile.reliability ==
+                QoSReliabilityPolicy.RELIABLE):
+                    reliability_reliable_endpoints_count += 1
+            if (info.qos_profile.durability ==
+                QoSDurabilityPolicy.TRANSIENT_LOCAL):
+                    durability_transient_local_endpoints_count += 1
+
+        if publishers_count == 0:
+            return qos_profile
+
+        # If all endpoints are reliable, ask for reliable
+        if reliability_reliable_endpoints_count == publishers_count:
+            qos_profile.reliability = QoSReliabilityPolicy.RELIABLE
+        else:
+            if reliability_reliable_endpoints_count > 0:
+                print(
+                    'Some, but not all, publisher are offering '
+                    'QoSReliabilityPolicy.RELIABLE. Falling back to '
+                    'QoSReliabilityPolicy.BEST_EFFORT as it will connect '
+                    'to all publishers', end='\n\n'
```

Comment:
is there any benefit of using `end`?
do we need to print this message?
should we use stderr?

```suggestion
                    'to all publishers\n\n'
```

Reply:
Addressed in: [afb4873
](https://github.com/gonzodepedro/ros2cli/commit/afb4873b11076b8a1a4f80f01a8c724ae97dc197)